aptos-move/block-executor/src/view.rs

```diff
@@ -133,6 +143,118 @@ impl<'a, T: Transaction, X: Executable> ParallelState<'a, T, X> {
         }
     }
 
+    fn read_from_group(
+        &self,
+        group_key: &T::Key,
+        resource_tag: &T::Tag,
+        txn_idx: TxnIndex,
+    ) -> anyhow::Result<GroupReadResult> {
+        use MVGroupError::*;
+
+        if let Some(DataRead::Versioned(_, v)) =
+            self.captured_reads
+                .borrow()
+                .get_by_kind(group_key, Some(resource_tag), ReadKind::Value)
+        {
+            return Ok(GroupReadResult::Value(v.extract_raw_bytes()));
+        }
+
+        loop {
+            match self
+                .versioned_map
+                .group_data()
+                .read_from_group(group_key, resource_tag, txn_idx)
+            {
+                Ok((version, v)) => {
+                    let data_read = DataRead::Versioned(version, v.clone());
+
+                    assert_ok!(
+                        self.captured_reads.borrow_mut().capture_read(
+                            group_key.clone(),
+                            Some(resource_tag.clone()),
+                            data_read
+                        ),
+                        "Resource read in group recorded once: may not be inconsistent"
+                    );
+
+                    return Ok(GroupReadResult::Value(v.extract_raw_bytes()));
+                },
+                Err(Uninitialized) => {
+                    return Ok(GroupReadResult::Uninitialized);
```

Comment:
This function is currently reading data only from versioned_map. If the data is not intialized in versioned_map, we should fetch the data from storage right?

Reply:
get_resource_from_group calls this, and then initialize_group to initialize.